lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4129,13 +4129,31 @@ AdjointEmitter::accumulateAdjointsDirect(AdjointValue &&lhs,
     // x + (y, z) => (x.0 + y, x.1 + z)
     case AdjointValueKind::Aggregate:
       SmallVector<AdjointValue, 8> newElements;
-      for (auto i : range(rhs.getNumAggregateElements())) {
-        auto lhsElt = builder.createTupleExtract(lhsVal.getLoc(), lhsVal, i);
-        auto rhsElt = rhs.takeAggregateElement(i);
-        newElements.push_back(accumulateAdjointsDirect(
-            makeConcreteAdjointValue(
-                ValueWithCleanup(lhsElt, lhsVal.getCleanup())),
-            std::move(rhsElt)));
+      auto lhsTy = lhsVal.getValue()->getType().getASTType();
```

Comment:
```suggestion
      auto lhsTy = lhsVal.getSwiftType();
```

Reply:
That's what I tried at first, but `lhsVal` is a `ValueWithCleanup` and doesn't have a `getSwiftType` member.

Reply:
You can use `lhs.getSwiftType()`. But no big deal.